lib/rewards/server/message-actions.js

```diff
@@ -11,46 +11,77 @@ class RewardMessageActions
+
+        }
+        await this.addItemToInventory(rewardObject.rewardModel, playerSchema);
+        this.removeDrop(rewardObject, room, client);
     }
 
-    removeDrop(rewardId, room, client)
+    removeDrop(rewardObject, room, client)
     {
-        let rewardObject = room.roomWorld.objectsManager.roomObjects[rewardId];
-        if (!rewardObject) {
-            return false;
-        }
         rewardObject.objectBody.world.removeBody(rewardObject.objectBody);
-        delete room.roomWorld.objectsManager.roomObjects[rewardId];
-        this.sendRemoveRewardMessage(rewardId, client);
+        let rewardObjectId = rewardObject.rewardModel.randomRewardId;
+        delete room.roomWorld.objectsManager.roomObjects[rewardObjectId];
+        this.sendRemoveRewardMessage(rewardObjectId, client);
     }
 
-    sendRemoveRewardMessage(rewardId, client)
+    sendRemoveRewardMessage(rewardObjectId, client)
     {
         // @TODO - BETA - Add new event here for the data.
         client.send('*', {
             act: RewardsConst.REWARDS_PICK_UP_ACT,
-            id: rewardId
+            id: rewardObjectId
         });
     }
 
-    async addItemToInventory(rewardId, room, playerSchema)
+    getRewardObject(rewardObjectId, room)
     {
-        let rewardObject = room.roomWorld.objectsManager.roomObjects[rewardId];
+        let rewardObject = room.roomWorld.objectsManager.roomObjects[rewardObjectId];
         if (!rewardObject) {
             return false;
         }
-        let rewardModel = rewardObject.rewardModel;
+        console.log('rewardObject found');
+        return rewardObject;
+    }
+
+    async addItemToInventory(rewardModel, playerSchema)
+    {
         let item = await playerSchema.skillsServer.dataServer.getEntity('item').loadById(rewardModel.itemId);
         let dropQty = rewardModel.dropQuantity;
         let rewardItem = playerSchema.inventory.manager.createItemInstance(item.key, dropQty);
         return await playerSchema.inventory.manager.addItem(rewardItem);
     }
+
+    isPlayerInsideDropInteractionArea(playerSchema, rewardObject, room)
+    {
+        // TODO: The interaction area should be defined on the rewardObject.
+        rewardObject.interactionArea = this.createInteractionAreaObject(room, rewardObject.x, rewardObject.y);
+        return rewardObject.interactionArea.isValidInteraction(
+            playerSchema.physicalBody.bodyState.x,
+            playerSchema.physicalBody.bodyState.y
+        );
+    }
+
+    createInteractionAreaObject(scene, x, y)
+    {
+        let interactionArea = new InteractionArea();
+        interactionArea.interactionArea = this.getRewardsInteractionDistanceConfig(scene);
+        interactionArea.setupInteractionArea(false, x, y);
+        return interactionArea;
+    }
+
+    getRewardsInteractionDistanceConfig(scene)
+    {
+        return scene.config.get('server/rewards/actions/interactionsDistance') ||
+            (scene.roomWorld.mapJson.tilewidth * 0.5);
```

Comment:
not sure if I did it on the other case, but lets put a DB config for the "0.5", also with a `GameConst.DEFAULT.TILE_PORTION` or something like that